api/server/router.go

```diff
@@ -9,99 +9,127 @@ import (
 	"net/http"
 	"sync"
 
-	"github.com/gorilla/mux"
```

Comment:
As part of this PR I decided to just gut our dependency on `gorila/mux`... I can remove it from the diff but we had to touch stuff with the router anyways... so it seemed forgivable....